src/riak_kv_qry_compiler.erl

```diff
@@ -123,7 +123,18 @@ compile_group_by(_, [], Acc, Q) ->
 compile_group_by(Mod, [{identifier,FieldName}|Tail], Acc, Q)
         when is_binary(FieldName) ->
     Pos = Mod:get_field_position([FieldName]),
-    compile_group_by(Mod, Tail, [{Pos,FieldName}|Acc], Q).
+    compile_group_by(Mod, Tail, [{Pos,FieldName}|Acc], Q);
+compile_group_by(Mod, [{time_fn,{_,FieldName},_} = GroupTimeFnAST|Tail], Acc, Q) when is_binary(FieldName) ->
+    GroupTimeFn = make_group_by_time_fn(Mod, GroupTimeFnAST),
+    compile_group_by(Mod, Tail, [{GroupTimeFn,FieldName}|Acc], Q).
+
+-spec make_group_by_time_fn(module(), group_time_fn()) -> function().
+make_group_by_time_fn(Mod, {time_fn, {identifier,FieldName},{integer,GroupSize}}) ->
+    Pos = Mod:get_field_position([FieldName]),
```

Comment:
I don't see any handling in tests or in code for specification of a field that does not exist. An error should be raised if the time function field parameter does not exist in the table.